### PR TITLE
 Refactor handling of WebSocket connections

### PR DESCRIPTION
**Pull request by lpinca**
- [x] This is a **bugfix**
- [x] This is a **code refactor**
- [ ] This is a **test update**
- [ ] This is a **typo fix**
- [ ] This is a **metadata update**

### For Bugs and Features; did you add new tests?

No.

### Motivation / Use-Case

First commit simplifies handling of WebSocket connections making sure that all connected clients receives data not just the last one.

Second commit prevents the process from exiting when an 'error' event is emitted on the WebSocket server client.

### CODE REVIEW

File: lib/DevServer.js [hunk 1]
```diff
@@ -116,67 +114,33 @@ module.exports = class DevServer extends EventEmitter {
     const { log, wss } = this;
     const { host, port, socket } = this.options;
 
-    const returnValue = this.server.listen(socket || port, host, (err) => {
+    const returnValue = this.server.listen(socket || port, host, () => {
```
Comment (lpinca) [inline lib/DevServer.js]: As far as I know the `'listening'` event handlers receive no arguments.

**Comment by lpinca** [comment]
Closing as per https://github.com/webpack/webpack-dev-server/issues/1245#issuecomment-356278199.

**Comment by shellscape** [comment]
@lpinca thanks for submitting the PR none the less. I'm sorry that I didn't get a chance to review it. I'm currently the only active maintainer and I'm getting married in a few days, so it's been difficult to carve out time for the project. Cheers 🍻

**Comment by lpinca** [comment]
No problem and congratulations 🎉 .

**Comment by lpinca** [comment]
@qiulang I don't know, what you are describing sounds like a different issue. This patch had nothing to do with Socket.IO.

**Pull request closed without merging**